standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -1454,6 +1454,16 @@ public void createTable(CreateTableRequest request) throws
       request.setProcessorIdentifier(processorIdentifier);
     }
 
+    EnvironmentContext envContext = request.getEnvContext();
+    envContext = Optional.ofNullable(envContext).orElse(new EnvironmentContext());
+
+    Map<String, String> props = Optional.ofNullable(envContext.getProperties()).orElse(new HashMap<>());
+    props.put(StatsSetupConst.HIVE_ICEBERG_STATS_SOURCE, 
+            conf.get(StatsSetupConst.HIVE_ICEBERG_STATS_SOURCE, HiveMetaHook.ICEBERG.toLowerCase()));
+
+    envContext.setProperties(props);
```

Comment:
does ``envContext.putToProperties()`` work?

Reply:
Yes, it is a good improvement. Done.